### DISCUSSION TITLE
Support supplying energymeter data via the API

### DISCUSSION THREAD

**Discussion by daninden** [serkri/SmartEVSE-3#134] [Ideas]
Currently you can select the API for supplying mains meter data (L1-L3). Would it be possible to support this also for the energymeter?

Looks like it's a bit more data:
import energy
export energy
power 
current L1-L3

**Comment by HA-TB303** [comment]
That would be nice, In that case I would be able to install the meter in the fusebox in stead of outside in the charger. Then feed the information from esphome or Home Assistant to the evse.

**Comment by dingo35** [comment]
If by energymeter you mean EVmeter, that stuff is already there. Press the "raw data" button...
If you dont see ev_meter import_active_energy and export_active_energy there, you are on an old version.

**Comment by daninden** [comment]
i don't mean getting the data.... i mean supplying it.
I started a branch for this:
https://github.com/daninden/SmartEVSE-3/tree/feature/em_api_support

**Comment by dingo35** [comment]
Ok so its a duplication of #133 .

**Comment by daninden** [comment]
well... sort of.. thats an Issue ;) (and therefore didn't see it) this is a feature request :) but funcationally yes

**Comment by daninden** [comment]
i will test my code.. if it works.. create a PR

**Comment by dingo35** [comment]
Dan Im still missing a use case for this ...

**Comment by daninden** [comment]
the exact same use case as the mainsmeter.. for meters that don't have modbus or cable too far

**Comment by dingo35** [comment]
1 Well if you are going to buy a SmartEVSE you probably dont have an EVmeter yet; so why buy one with an inferior interface?
This is different from a Mainsmeter, most people have one already :-) and you usually cannot choose the interface it provides...

2. Cable too far: the logical place to install an EVmeter is right next to the SmartEVSE, but even if you choose to install it in your main power panel you would want to connect it with SmartEVSE through reliable modbus

3. SmartEVSE does _nothing_ with the EV meter info, except when you have MaxCircuit activated (and in that case you need reliable data feed). In all other cases it just displays the data on the display, the webserver and the API. Why would you pump data around when you already have it in your domotica system?

**Comment by daninden** [comment]
I already had an evmeter, i want to use MaxCircuit...
if API is not reliable, why allow it for mainsmeter then?

well.. it wasn't much work, created a PR
https://github.com/serkri/SmartEVSE-3/pull/135

you can always reject it ;)

**Reply by dingo35** [reply]
Because most existing Mainsmeters dont have any choice on their interface, that was allowed. And it gives a lot of trouble, people feeding it with frequency too low, they get timouts, they solve that by replicating and resending old values; their feeding scripts dont work, they dont know how to detect and handle negative values.

Take @HA-TB303 's use case: he can take the modbus route, or the EM_API if we enable this. It makes no sense to make a long communication route with multiple error possibilities, if the modbus, exactly designed for these kinds of comms is there.
So on one hand there are a few users (like you) that want to reuse an already bought EV meter, on the other hand there is a large group of new users who need a stable architecture to guide them in their purchase / implementation decisions...

**Reply by daninden** [reply]
Also for mainsmeter ppl have a choice (the sensorbox, or any other modbus meter), it's just often more conveniant to go the API way.
I think this fork has gotten it's popularity due to the API support.
I do understand why you would like ppl to take a certain route, just don't agree on making that the only route.
